### ISSUE TITLE
[a11y]: Providing options other than use of lists for accordion components

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#9853]
### Package

carbon-components

### Browser

Chrome

### Operating System

MacOS

### Package version

whatever is used on carbondesignsystem.com 

### Automated testing tool and ruleset

keyboard only

### Assistive technology

JAWS

### Description

On a page, there are a number of single accordions titled "Why" 

The default construction of an accordion from carbon is to put it in an unordered list. That may make sense for a set of accordions, but makes less sense for 1 accordion: JAWS announces that as 'list of one items'!!
![image.png](https://zenhub.ibm.com/images/5cab73f442b8dc5ede7b84cc/f305a5e0-c04f-4609-b6e4-2511cbb7de76)

It would be good to either give teams the ability to turn off the enumeration of accordion sections, or to only put them in a list when there is more than one item.
Currently using `<ul>` and `<li>` as part of code. Suggest an option to replace these with divs, which should still allow all the function of the accordion without it being put into a list.

### WCAG 2.1 Violation

No, but lists are not used in the Aria authoring practices examples for accordions

### CodeSandbox example

n/a

### Steps to reproduce

1. Locate the Why sections on https://www.ibm.com/able/toolkit/plan/project
2. Turn on JAWS or other screen reader
3. navigation to the Whe section; JAWS will announce "list of 1 items"

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by joshblack** [comment]
Should we just always default to a div for these components @mbgower? Or would we need to support both cases (list or divs) with a reasonable default (maybe div as the default?)

**Comment by mbgower** [comment]
@joshblack returning to this issue as I capture all considerations for accordion from my reviews and updates.
I suggest investigating the following:
- remove the list by default for accordions
- make the accordion title be a heading by default
- where implementers choose to _not_ make accordion titles headings, put the accordions in a list

Are there technical problems with this approach? I'm going to open up a discussion on accordions having headings by default. I'll cross-reference this.
